### ISSUE TITLE
Unused format string literals (string interpolation)

### ISSUE THREAD

**Issue by alice-i-cecile** [rust-lang/rust-clippy#13651]
### What it does

When the user attempts to use `{}`, `{:?}` or similar invocations, we should expect them to actually have a value. Warn the user when no value is provided and "{}" or the like would actually be printed.

Inspired by https://github.com/bevyengine/bevy/pull/16228

### Advantage

These errors are likely an oversight, and they should either be removed or the correct value added.

### Drawbacks

It will make it harder to draw strange ASCII faces in your error messages.

### Example

```rust
            .expect("Component id: {:?}, already has an on_add hook")
```

Could be written as:

```rust
            .expect("Component already has an on_add hook")
```

**Comment by Piturnah** [comment]
The problem here, as I understand it, is that `expect` takes a `&str`, not a format string. So you actually couldn't supply a value. 

In my opinion, a lint should be added for `&str` literals which include format-string syntax (like `{:?}`) to make sure the user actually intended the literal string "{:?}" or whether they thought they were in a format string and forgot a value.

Hopefully it's clear what I mean...

**Comment by d-e-s-o** [comment]
> In my opinion, a lint should be added for `&str` literals which include format-string syntax

I think that's covered by https://github.com/rust-lang/rust-clippy/issues/10195
